adbackend/src/main/java/com/csc/adbackend/CampaignServiceImpl.java

```diff
@@ -132,5 +133,26 @@ public void updateCampaign(Integer campId, Campaign campaign) {
         return responseEntity;
     }
 
+    public ResponseEntity<String> updateAd(Integer campId, Integer adId, Ad ad) {
+        ResponseEntity<String> responseEntity;
+
+        Optional<Campaign> camp = campaignRepo.findById(campId);
+        if (camp.isPresent()) {
+            try {
+                camp.get().getAds().remove(adId);
+                camp.get().getAds().put(adId, ad);
+                campaignRepo.save(camp.get());
+                responseEntity = new ResponseEntity<>("Ad Updated.", HttpStatus.OK);
+
+            } catch (IllegalArgumentException e) {
+                responseEntity = new ResponseEntity<>("Ad not found.", HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
According to spec, should return 400 error with body of the form {"error":"notFound", "details":[]}